### ISSUE TITLE
ResourcePrinicipal authentication forces primary region always (unable to override)

### ISSUE THREAD

**Issue by cweeks72** [oracle/terraform-provider-oci#1761]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

``` Terraform v1.2.9 on linux_amd64```
```OCI version: 4.106.0```

### Affected Resource(s)

affected_resources = terraform

### Terraform Configuration Files

```hcl
provider "oci" {
  auth = "ResourcePrincipal"
  region = "uk-cheesebread-1"
}

data "oci_objectstorage_namespace" "os_namespace" {
}
```

### Debug Output

The above code works, but it shouldn't because the region `uk-cheesebread-1` does not exist. It always uses the region the code runs on, which can be problematic if you're trying to push changes to another region from OCI devops. I suspect this is also a problem in ResourceManager systems which are trying to do cross-regional stuff as well. 

### Expected Behavior

The code should have failed because the region is invalid.

### Actual Behavior

The code worked and accessed the region where the code is running.

### Steps to Reproduce

1. Create a devops project build pipeline.
2. Build pipeline should run `terraform init`
3. Then it should run `terraform apply` (see build_spec.yaml file)

Example build_spec.yaml file:

```yaml
version: 0.1
component: build
timeoutInSeconds: 10000
shell: bash
failImmediatelyOnError: true
env:
  variables:
    "OCI_GO_SDK_DEBUG": "v"
    "TF_LOG": "DEBUG"
steps:
  - type: Command
    name: Init terraform
    command: terraform init -no-color
  - type: Command
    name: Apply terraform
    command: terraform apply -no-color -auto-approve
```

When you run this, it should FAIL with the terraform specified above, since the region is non-existent. It does not, and instead accesses the region where the OCI devops code is running.

### Important Factoids

I am trying to automate some terraform activities using OCI devops and also Resource Manager. I have multiple regions I might need to access (ashburn for prod, phoenix for dev) from a terraform activity running in a single region, and I noticed that I cannot access a region outside the one where my code is running.

Looking at the code https://github.com/oracle/terraform-provider-oci/blob/fbc2100febe63ac8ac047709f3db1d953f1bfb21/vendor/github.com/oracle/oci-go-sdk/v65/common/auth/resouce_principal_key_provider.go, it seems that you ALWAYS force the region to the `OCI_RESOURCE_PRINCIPAL_REGION` value from the resource principal API, the region key specified in the provider block seems to be completely ignored.

I tested the OCI CLI in OCI devops and it works and can access the remote region fine by using the --region override.

Finally, as a test for terraform, I forced the value of the `OCI_RESOURCE_PRINCIPAL_REGION` environment variable in OCI devops, and was able to set the region to something else. Sadly, this won't work for ResourceManager based terraform deployments, since you cannot control the environment there.

**Comment by ravinitp** [comment]
Thank you for reporting the issue. We observed the affected resources are not provided in the description or it's incorrect. We request you to add it in issue description as mentioned in below format.
Example: affected_resources = oci_core_instance , oci_core_instances
 
If it's not related to any particular resource then mention affected resource as terraform.
Example: affected_resources = terraform
 
As this works through automation, request you to follow exact syntax.

**Comment by ravinitp** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by NagendraNigade** [comment]
Hi,
Resolution for this issue has been released in Terraform provider OCI version 4.118.0
Thanks.
